Packages/SmithyTestUtil/Sources/XCTAssertions.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+public func XCTAssertJSONDataEqual(
+    _ expression1: @autoclosure () throws -> Data,
+    _ expression2: @autoclosure () throws -> Data,
+    _ message: @autoclosure () -> String = "",
+    file: StaticString = #filePath,
+    line: UInt = #line
+) {
+    do {
+        let data1 = try expression1()
+        let data2 = try expression2()
+        guard data1 != data2 else { return }
+        let jsonDict1 = try JSONSerialization.jsonObject(with: data1) as! NSDictionary
+        let jsonDict2 = try JSONSerialization.jsonObject(with: data2) as! NSDictionary
```

Comment:
Why cast to `NSDictionary` here; why not keep it as a Swift type?

Also, are we sure we will never be comparing JSON arrays or fragments here?

Reply:
> Why cast to NSDictionary here; why not keep it as a Swift type?

`NSDictionary` gives us equality checking for free. The bridged swift type would require explicit typing or require a custom algorithm to do a deep equality check. 

> Also, are we sure we will never be comparing JSON arrays or fragments here?

Currently yes given all the currents tests pass.
I'd prefer to defer adding additional handling until it is needed in an effort to not write code that is never used. (adding handling for arrays would be simple but supporting fragments could be complicated. Currently `JSONSerialization` will fail to even generate an object for fragments given we aren't passing in [the option to allow fragments](https://developer.apple.com/documentation/foundation/jsonserialization/writingoptions/3081259-fragmentsallowed))